File: tests/cpp/Animals.cpp
class Animal {
public:
    std::string name;

    Animal(std::string name) : name(name) {}

    void speak() {
        // Animal speaks
    }
};

class Dog : public Animal {
public:
    Dog(std::string name) : Animal(name) {}

    void speak() {
        // Dog barks
    }
};

class Cat : public Animal {
public:
    Cat(std::string name) : Animal(name) {}

    void speak() {
        // Cat meows
    }
};

File: tests/java/Animals.java
public class Animal {
    private String name;

    public Animal(String name) {
        this.name = name;
    }

    public void speak() {
        System.out.println("Animal speaks");
    }
}

class Dog extends Animal {
    public Dog(String name) {
        super(name);
    }

    @Override
    public void speak() {
        System.out.println("Dog barks");
    }
}

class Cat extends Animal {
    public Cat(String name) {
        super(name);
    }

    @Override
    public void speak() {
        System.out.println("Cat meows");
    }
}
